sdk/create-dapp/templates/react-e2e-counter/src/CreateCounter.tsx

```diff
@@ -10,33 +12,12 @@ export function CreateCounter({
 }) {
   const counterPackageId = useNetworkVariable("counterPackageId");
   const suiClient = useSuiClient();
-  const { mutate: signAndExecute } = useSignAndExecuteTransaction({
-    execute: async ({ bytes, signature }) =>
-      await suiClient.executeTransactionBlock({
-        transactionBlock: bytes,
-        signature,
-        options: {
-          // Raw effects are required so the effects can be reported back to the wallet
-          showRawEffects: true,
-          showEffects: true,
-        },
-      }),
-  });
-
-  return (
-    <Container>
-      <Button
-        size="3"
-        onClick={() => {
-          create();
-        }}
-      >
-        Create Counter
-      </Button>
-    </Container>
-  );
+  const { mutate: signAndExecute } = useSignAndExecuteTransaction();
+  const [waitingForTxn, setWaitingForTxn] = useState(false);
```

Comment:
The useSignAndExecuteTransaction already has state for this, and returns the following properties that can be used:

`isIdle`, `isPending`, `isSuccess`, `isError`, and `status`.

You probably want `isPending`

Reply:
sorry, I misremembered how this works.  onSuccess is not included in these statuses.

Reply:
Tried this. The issue is that waitingForTxn is `true` until after the transaction is waited for. `isPending` becomes false after the transaction is executed, not waited for though. 

### with waitingForTxn
https://github.com/user-attachments/assets/8c9b25a9-9f8e-4c11-8f8d-1d5e862e24f0

### with isPending

https://github.com/user-attachments/assets/eee2ea0a-3a90-48c6-8fe1-d923f957a37d

Reply:
`isSuccess` should work though

Reply:
I ended up using `isSuccess` and `isPending` since when I just used `isSuccess`, the spinner won't show after the button is pressed but before the user accepts the transaction.